### ISSUE TITLE
Only public cloud infrastructure?

### ISSUE THREAD

**Issue by pinterb** [kris-nova/cluster-api#2]
Is this proposal for public cloud infrastructure only?  In other words, is on premises bare metal and virtualization to be excluded?  If so, I think that distinction needs to be made clear.

**Comment by kris-nova** [comment]
I see no reason we can't attempt to encapsulate baremetal infrastructure in this API as well.. 
The only nuance being that a public cloud gives us a much more consumable way of representing infrastructure (by using their HTTP APIs)... but you have a +1 from me on supporting baremetal as well..

Did you see something that made you think otherwise?

**Comment by pinterb** [comment]
No, not really.  It's just that this effort IMO, is ambitious.  If there were any concerns about "boiling the ocean", one way to mitigate those concerns would be to limit the initial scope (e.g. to public cloud).  So basically, I was just looking for clarity on the scope of this effort.  Thanks.
